### ISSUE TITLE
Find Encryption Keys displaying keys with non-matching UIDs

### ISSUE THREAD

**Issue by bnvk** [mailpile/Mailpile#1542]
Perhaps relating to #1539 using the same key lookup feature- some lookups actually showing other peoples keys which non-matching UIDs (email addresses). It seems to be when the following conditions are met:

- Looking up an name@somedomain.org address
- Other keys that also have matching @somedomain.org UIDs
- One of more of these keys are signed by these other @somedomain.org addresses

I can share which keys these are if you want to try replicating it at some point!

**Comment by BjarniRunar** [comment]
Can you tell me what the Mailpile key scoring info says about such keys?  Are they found on the keychain, or are they being found in e-mail, or keyservers...?

**Comment by bnvk** [comment]
@BjarniRunar yes, all of the keys are in the keychain already. The first key, which in this case is the only key which should be showing says:

```
Score +55
Encryption key is very strong +4
Found encryption key in keychain +1
Encryption key has been imported and verified +50
```

Incidentally, it's actually showing my key, albeit a weird "version" of my key. One which is missing a few recently added UIDs
